src/module_1/module_1_meteo_api.py

```diff
@@ -9,29 +12,87 @@
 }
 VARIABLES = ["temperature_2m_mean", "precipitation_sum", "wind_speed_10m_max"]
 START_DATE = "2010-01-01"
-END_DATE = "2020-12-31"
+END_DATE = "2020-01-02"
+
+
+def get_data_from_api(url: str, params: dict[str, Any]) -> Any:
```

Comment:
We normally do more than one call to the API. Imagine you get a denial because a network issue that lasts some micro-seconds. We call that retry, and we have to be smart about it. If you don't know about what exponential backoff is, take a look at it.

Reply:
Ok. I have had no real experience working with APIs. I'll try to apply what Guille said us in the lesson to improve the function.